pallets/utility/src/lib.rs

```diff
@@ -187,10 +187,6 @@ decl_module! {
         #[weight = <T as Trait>::WeightInfo::batch(&calls)]
         pub fn batch(origin, calls: Vec<<T as Trait>::Call>) {
             let is_root = ensure_root(origin.clone()).is_ok();
-            if !is_root {
-                let sender = ensure_signed(origin.clone())?;
```

Comment:
If it's not root, we should call the `ensure_signed` function. Don't need to check permissions but it's best to catch any unsigned tx asap. Same comment for other extrinsics.